src/librc/librc.c

```diff
@@ -596,6 +596,24 @@ rc_is_user(void)
 	return is_user;
 }
 
+int scriptdirfds[SCRIPTDIR_MAX];
+
+int rc_scriptdirfd(const char * const *scriptdir) {
+	const char * const *scriptdir_array = is_user ? rc_dirs.scriptdirs : scriptdirs;
+	size_t idx;
+
+	if (!scriptdir || !*scriptdir || scriptdir < scriptdir_array)
```

Comment:
Comparing potentially unrelated pointers via relational operator is UB.
https://www.iso-9899.info/n3220.html#6.5.9p6